contracts/protocol/libs/EIP712.sol

```diff
@@ -130,15 +128,6 @@ contract EIP712 is SignatureErrors {
 
                     return;
                 }
-            } else {
-                if (returnData.length == 0) {
-                    revert SignatureValidationFailed();
-                } else {
-                    /// @solidity memory-safe-assembly
-                    assembly {
-                        revert(add(SLOT_SIZE, returnData), mload(returnData))
```

Comment:
Return data could still have something useful inside, so I wouldn't just disregard it.

I would actually restore this code and handle the whole issue a bit differently.

I'change the line 113 from `if (_user.code.length > 0)` to `if (_user.code.length > 0 && sig.v == 0)`
If the caller provided non-zero signature elements, it suggests they also provided a (valid) signature and we should not try to verify contract signature.  
This solves the issue, makes code more efficient and bubble up revert reasons if there are any.